designs/compatibility/arm64-support.md

```diff
@@ -0,0 +1,150 @@
+
+### Goals
+
+- **Multi-Architecture Container Images**: Provide official `arm64` container images for all OpenEBS control-plane and data-plane components.
+- **CI Validation and Testing**: Integrate `arm64` tests into the existing CI pipelines to ensure feature parity, performance metrics validation, and interoperability.
+- **Documentation and Guidance**: Update official documentation to clearly state which architectures are supported and provide guidance on deploying OpenEBS clusters on `arm64` environments.
+- **Community Feedback Loop**: Establish channels (e.g., Slack, GitHub discussions, mailing lists) to gather early feedback from users running on `arm64` and continuously improve support.
+
+### Non-Goals
+
+- **Architecture-Specific Optimizations**: This OEP does not focus on deep, architecture-specific optimizations tailored for `arm64` (e.g., using ARM-specific CPU instructions). The goal is baseline parity, not heavily optimized builds.
+- **Retroactive Support for Legacy Architectures**: We will not extend support to other architectures beyond `arm64` and `amd64`.
+- **Performance Guarantees**: While we aim for parity, we will not provide strict performance guaranatees specific to `arm64` at this stage.
+- **Immediate Full Feature Set Validation on `arm64`**: We will start with core components and eventually roll out to all official modules. Some pre-release/non-official engines (Local Rawfile) or features may come later.
+
+## Proposal
+
+This proposal involves extending current build and release workflows to produce `arm64` images and validate them through CI/CD. It includes adding `arm64` test runners, ensuring that Helm charts and manifests work on `arm64` clusters, and validating that all dependencies are functional on Arm architecture.
+
+### User Stories
+
+#### Story 1
+
+**As a developer** running a test Kubernetes cluster on my local M-series MacBook, I want to deploy OpenEBS to manage my local storage. Without official `arm64` images, I feel boxed out of developing on top of OpenEBS. Official support ensures I can quickly deploy and maintain OpenEBS on my machine without custom tooling, virtualization, or complicated workarounds.
+
+#### Story 2
+
+**As an operator** managing AWS Graviton-based EKS clusters, I need a storage solution that runs natively and efficiently on `arm64`. With official `arm64` support in OpenEBS, I can deploy OpenEBS alongside my other Arm-native applications, simplifying my environment and reducing operational overhead related to mixed-architecture maintenance.
+
+### Implementation Details/Notes/Constraints
+  
+- **Dependency Checks**:
+  - Verify that all external dependencies (`spdk-rs`, `bitnami-shell`, etc.) provide `arm64` binaries or have acceptable fallbacks.
+
+- **Control-plane Builds**:
+  - Update control-plane builds to support `arm64`.
+
+- **Data-plane build**:
+  - Update data-plane builds to support `arm64`.
+    - LocalPV-HostPath
+    - LocalPV-ZFS
+    - LocalPV-LVM
```

Comment:
I was basing my notes on [this diagram](https://openebs.io/docs/concepts/architecture), are `LocalPV-*` engines more control-plane related?

Reply:
hmm I think that diagram is a little confusing.
It's showing the data-engines connecting to the store, but I think here data-engines does not mean data-plane, since  for `LocalPV*`, the data-plane are the kernel LVM/ZFS drivers. It just means the "data-engine operators".
@niladrih @Abhinandan-Purkait @avishnu do we need to tweak those docs a bit?

Reply:
Probably should have been storage engines?